### ISSUE TITLE
many unimplemented flags in ssl_conf.c

### ISSUE THREAD

**Issue by richsalz** [openssl/openssl#10160]
The command table in ssl_conf.c (used by ```SSL_CONF_cmd```) has several options that are not implemented by the commands.  Should we remove them?
```
ecdh_single
anti_replay
no_anti_replay
cert
key
chainCApath
chainCAfile
verifyCApath
verifyCAfile
requestCAFile
num_tickets
```

**Comment by levitte** [comment]
I don't quite understand what you mean with them not being implemented...  Basically, *anything* defined in ssl_conf.c is useful to any program that calls `SSL_CONF_cmd_argv`, and well, a simple grep is revealing:

```
: ; git grep SSL_CONF_cmd_argv | grep -E -v '^(doc|util)/'
demos/bio/client-arg.c:        rv = SSL_CONF_cmd_argv(cctx, &nargs, &args);
demos/bio/server-arg.c:        rv = SSL_CONF_cmd_argv(cctx, &nargs, &args);
include/openssl/ssl.h:__owur int SSL_CONF_cmd_argv(SSL_CONF_CTX *cctx, int *pargc, char ***pargv);
ssl/ssl_conf.c:int SSL_CONF_cmd_argv(SSL_CONF_CTX *cctx, int *pargc, char ***pargv)
test/ssltest_old.c:            rv = SSL_CONF_cmd_argv(c_cctx, &argc, &argv);
test/ssltest_old.c:                rv = SSL_CONF_cmd_argv(s_cctx, &argc, &argv);
```

The thing I find weird, though, is that this isn't used by `s_server` or `s_client`, so we end up with duplicate code to do the same thing...

**Comment by richsalz** [comment]
> The thing I find weird, though, is that this isn't used by s_server or s_client, so we end up with duplicate code to do the same thing...

*that's* what I was complaining about.

The old ssltest and demo programs... not sure how much we care about this, and if any invocations use the above flags.

**Comment by levitte** [comment]
Thing is that s_client and s_server implement the same on their own. The smart thing would be to refractor them to use SSL_CONF_cmd_argv, no ? I suspect that was intended, but with the usual flurry of things happening, that never got beyond a private branch or something like that.

**Comment by richsalz** [comment]
Yes, that makes sense.  I will do it after all my other doc refactoring PR's get reviewed and merged (so, like, maybe a month from now? :)  There are too many moving parts to the document and code shuffling before then.

**Comment by richsalz** [comment]
No, it doesn't make sense :(  s_client, for example, implements both -CAFile and -chainCAFile (and xxxPath variants) and uses them differently, and the latter differently from the SSL_CONF api.

Sigh.

**Comment by mattcaswell** [comment]
> The thing I find weird, though, is that this isn't used by s_server or s_client, so we end up with duplicate code to do the same thing...

I'm confused. `SSL_CONF_cmd_argv` just calls `SSL_CONF_cmd` to process these arguments. And s_server and s_client call `SSL_CONF_cmd` directly to process them. So where in that do we have duplicate code?

**Comment by richsalz** [comment]
Look for any of the chain settings; certCAChain, etc.  They are used internally in the app

**Comment by richsalz** [comment]
Look for any of the chain settings; certCAChain, etc.  They are used internally in the app, not passed to the CONF routine.

**Comment by mattcaswell** [comment]
Ah, ok. I now understand. There are lots of options that *are* processed correctly via `SSL_CONF_cmd` but there are some that are not.

Note that from your original list `anti_replay`, `no_anti_replay` and `num_tickets` are all implemented correctly (no duplication).

**Comment by richsalz** [comment]
I will be taking a look at this after my other PR's get disposed of.  (Nudge, nudge. :)  Then I'll dig further into the code also.

**Comment by richsalz** [comment]
Some of the documentation is fixed in #10191.  At least the three "missing" ones that @mattcaswell mentioned in https://github.com/openssl/openssl/issues/10160#issuecomment-541612506 are now in the s_server doc.

As for the ones that aren't implemented by the commands, I split them off into their own head2 section within the ```SSL_CONF_cmd``` file.
